src/Core/Managed/Shared/Channel/IDeepCloneable.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.ApplicationInsights.Channel
+{
+    /// <summary>
+    /// Indicates that an object support deep cloning.
+    /// </summary>
+    /// <typeparam name="T">The type of the object that results from the cloning operation.</typeparam>
+    public interface IDeepCloneable<out T>
+    {
+        /// <summary>
+        /// Clones the object deeply, so that the original object and its clones share no state 
+        /// and can be modified independently.
+        /// </summary>
+        /// <returns>The cloned object.</returns>
+        T DeepClone();
```

Comment:
I want to make sure that every telemetry is cloanable. And Application Insights SDK will never have anything else defined cloanable. With such a small public surface I think it is a safe assumption. And what if somebody will take a dependency on this interface? Application Insights SDK is not a general purpose library with common classes and this sort of dependency will force us to keep the interface around